### PR TITLE
Container and Text components

### PR DESCRIPTION
**Pull request by hcarneiro**
- Added container and text components support
- Updated a toggle field with new sub-types

### CODE REVIEW

File: src/Application.vue [hunk 1]
```diff
@@ -82,10 +83,25 @@ export default {
...
+          && !state.widgetMode
+          && this.supportsContainers(configuration)
+      })
 
+      return configurations
+    }
   },
   methods: {
+    supportsContainers(configuration) {
+      if (typeof configuration.appSupportsContainers === 'undefined') {
```
Comment (tonytlwu) [inline src/Application.vue]:
```suggestion
      return typeof configuration.appSupportsContainers === 'undefined'
        || this.appSupportsContainers === configuration.appSupportsContainers
```

File: src/components/UI/ComponentSettings.vue [hunk 2]
```diff
@@ -163,7 +174,29 @@ export default {
 
       return false
     },
+    supportsContainers(configuration) {
+      // If not defined returns true to show
```
Comment (tonytlwu) [inline src/components/UI/ComponentSettings.vue]:
```suggestion
      // If not defined returns true to show
      return typeof configuration.appSupportsContainers === 'undefined'
        || this.appSupportsContainers === configuration.appSupportsContainers
```

File: src/components/fields/DisplayExtendedField.vue [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  },
+  components: {
+    InheritDot
+  },
+  props: {
+    data: Object
+  },
+  watch: {
+    value(newVal, oldVal) {
+      if (newVal != oldVal && !this.fromReset) {
```
Comment (tonytlwu) [inline src/components/fields/DisplayExtendedField.vue]:
```suggestion
      if (newVal !== oldVal && !this.fromReset) {
```

Comment (tonytlwu) [reply src/components/fields/DisplayExtendedField.vue]: There are plenty other instances in this PR

File: src/components/UI/ComponentSettings.vue [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <template>
   <transition :name="transition">
-    <div v-if="state.appearanceGroupOverlay && state.appearanceGroupOverlay.isOpen" id="component-settings-overlay">
+    <div v-if="state.appearanceGroupOverlay && state.appearanceGroupOverlay.isOpen" id="component-settings-overlay" ref="componentoverlay">
```
Comment (squallstar) [inline src/components/UI/ComponentSettings.vue]: is this correct that it's not in camel case? `componentOverlay`

File: src/components/fields/FlexAlignField.vue [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+  },
+  components: {
+    InheritDot
+  },
+  props: {
+    data: Object
+  },
+  watch: {
+    value(newVal, oldVal) {
+      if (newVal != oldVal && !this.fromReset) {
```
Comment (squallstar) [inline src/components/fields/FlexAlignField.vue]: `!==`

File: src/components/fields/ToggleField.vue [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+  props: {
+    data: Object
+  },
+  watch: {
+    valueToShow(newValue) {
+      const valueProperty  = _.find(this.properties, { valueToShow: newValue })
+      this.value = valueProperty.value
+    },
+    value(newVal, oldVal) {
+      if (newVal != oldVal && !this.fromReset) {
```
Comment (squallstar) [inline src/components/fields/ToggleField.vue]: !==

File: src/Application.vue (hunk 1 above)
Comment (hcarneiro) [reply src/Application.vue]: Updated

File: src/components/UI/ComponentSettings.vue (hunk 2 above)
Comment (hcarneiro) [reply src/components/UI/ComponentSettings.vue]: Updated

File: src/components/fields/DisplayExtendedField.vue (hunk 3 above)
Comment (hcarneiro) [reply src/components/fields/DisplayExtendedField.vue]: Updated all `!=` and `==` to `!==` and `===` respectively.

File: src/components/fields/FlexAlignField.vue (hunk 5 above)
Comment (hcarneiro) [reply src/components/fields/FlexAlignField.vue]: Updated

File: src/components/fields/ToggleField.vue (hunk 6 above)
Comment (hcarneiro) [reply src/components/fields/ToggleField.vue]: Updated

File: src/components/UI/ComponentSettings.vue (hunk 4 above)
Comment (hcarneiro) [reply src/components/UI/ComponentSettings.vue]: Updated to camelCase everywhere

**Pull request merged**